.github/workflows/test_spack.yaml

```
@@ -0,0 +1,44 @@
+name: k4GenCITests
+on: [push, pull_request]
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - uses: cvmfs-contrib/github-action-cvmfs@v2
+      with:
+        cvmfs_repositories: 'sw.hsf.org,sw-nightlies.hsf.org,sft-nightlies.cern.ch,sft.cern.ch,geant4.cern.ch'
+    - name: Start container
+      run: |
+        docker run -it --name CI_container -v ${GITHUB_WORKSPACE}:/Package -v /cvmfs:/cvmfs:shared -d clicdp/cc7-lcg /bin/bash
+    - name: Setup container
+      run: |
+        docker exec CI_container /bin/bash -c " ln -s /usr/lib64/liblzma.so.5.2.2 /usr/lib64/liblzma.so;"
+
+    - name: Setup Spack
+      run: |
+        docker exec CI_container /bin/bash -c 'cd ./Package;\
+        curl -L -o spack.tar.gz https://gitlab.cern.ch/key4hep/k4-deploy/-/jobs/artifacts/release/raw/key4hep-spack_centos7-cvmfs.tar.gz?job=setup_spack_push;\
+        tar xfz spack.tar.gz && rm spack.tar.gz;'
+    - name: Compile
+      run: |
+        docker exec CI_container /bin/bash -c 'cd ./Package; 
+        source spack/share/spack/setup-env.sh;\
+        spack dev-build k4simdelphes@main;' # --test=all needs some work
```

```suggestion
        source spack/share/spack/setup-env.sh;\
        spack install --only dependencies k4simdelphes@main; \
        spack dev-build k4simdelphes@main;' # --test=all needs some work
```
Could we "declutter" the output of this CI workflow by installing the dependencies first in a slightly less verbose way and only have the verbose output for the `dev-build` of the actual package?

---

Hmmm, in principle no dependencies should be installed, that's a temporary issue with the spack installation. I think the release workflow misses a test that the dependencies are picked up properly.

In general unfortunately, this spack based ci config does not work as well as I hoped right now. Compared to just sourcing the setup and running make, it is quite slower (even accounting for concretization), and sometimes runs out of disk space. I think I'll make a separate repo for that to investigate those issues properly.